nova_vm/src/ecmascript/builtins/control_abstraction_objects/promise_objects/promise_prototype.rs

```diff
@@ -127,36 +125,58 @@ impl PromisePrototype {
 pub(crate) fn perform_promise_then(
     agent: &mut Agent,
     promise: Promise,
-    on_fulfilled: Option<Function>,
-    on_rejected: Option<Function>,
+    on_fulfilled: Value,
+    on_rejected: Value,
     result_capability: Option<PromiseCapability>,
 ) {
     // 3. If IsCallable(onFulfilled) is false, then
     //     a. Let onFulfilledJobCallback be empty.
     // 4. Else,
     //     a. Let onFulfilledJobCallback be HostMakeJobCallback(onFulfilled).
+    // TODO: Add the HostMakeJobCallback host hook. Leaving it for later, since in implementations
```

Comment:
**thought:** I guess this could eventually be a build flag, so only browsers pay this price.

Reply:
Are we planning of implementing wasm enabled by default, without a build flag? Because it seems like the wasm JS API depends on the incumbent realm, which is an HTML spec concept which needs Host{Make,Call}JobCallback to set up. V8 implements those host hooks according to the HTML spec by itself, and it doesn't give the embedder even the decision to not implement them.

Reply:
Although I'm not sure if this is actually observable in any way in Node.js and Deno...